lib/omniauth/strategies/shopify.rb

```diff
@@ -96,6 +114,9 @@ def callback_phase
         unless valid_scope?(token)
           return fail!(:invalid_scope, CallbackError.new(:invalid_scope, "Scope does not match, it may have been tampered with."))
         end
+        unless valid_permissions?(token)
+          return fail!(:invalid_permissions, CallbackError.new(:invalid_permissions, "Requested permission level does not match."))
```

Comment:
Should we use "API access mode" instead of "permission level" to line up with our documentation?